tests/openedx_tagging/core/tagging/test_api.py

```diff
@@ -188,3 +194,205 @@ def test_tag_object(self):
+        self.assertIsNotNone(item)
+
+        # Check parent update
+        item = self.get_tag(tags, 'Eubacteria 2.0')
+        self.assertIsNotNone(item)
+
+        # Check parent deletion
+        self.assertEqual(item.parent.value, 'Archaea')
+        item = self.get_tag(tags, 'Archaebacteria 2.0')
+        self.assertIsNotNone(item)
+        self.assertIsNone(item.parent)
+
+        # Check new tag
+        item = self.get_tag(tags, 'My new tag 2.0')
+        self.assertIsNotNone(item)
+        self.assertEqual(item.parent.value, 'Bacteria 2.0')
+
+        # Check existing tag
+        item = self.get_tag(tags, 'Porifera')
+        self.assertIsNotNone(item)
+
+    def test_import_tags_validations(self):
+        invalid_csv_data = "id,name,tag_parent_id,tag_parent_name\n1,Tag 1,,\n2,Tag 2,1,Tag 1\n"
+        taxonomy = tagging_api.create_taxonomy("Validations_Taxonomy")
+        taxonomy_free_text = tagging_api.create_taxonomy("Free_Taxonomy", allow_free_text=True)
+
+        # Open the file in each test since it always closes even if there is an error
+        with self.assertRaises(ValueError):
+            invalid_csv_file = BytesIO(invalid_csv_data.encode())
+            tagging_api.import_tags(taxonomy_free_text, invalid_csv_file, tagging_api.TaxonomyDataFormat.CSV)
+
+        with self.assertRaises(ValueError):
+            invalid_csv_file = BytesIO(invalid_csv_data.encode())
+            tagging_api.import_tags(taxonomy, invalid_csv_file, "XML")
+
+        with self.assertRaises(ValueError):
+            invalid_csv_file = BytesIO(invalid_csv_data.encode())
+            tagging_api.import_tags(taxonomy, invalid_csv_file, tagging_api.TaxonomyDataFormat.CSV)
+
+    @ddt.data(
+        # Invalid json format
+        {"taxonomy_tags": [
+            {"id": "1", "name": "Tag 1"},
+            {"id": "2", "name": "Tag 2", "parent_id": "1", "parent_name": "Tag 1"}
+        ]},
+        # Invalid 'id' key
+        {"tags": [
+            {"tag_id": "1", "name": "Tag 1"},
+        ]},
+        # Invalid 'name' key
+        {"tags": [
+            {"id": "1", "tag_name": "Tag 1"},
+        ]}
+    )
+    def test_import_tags_json_validations(self, json_data):
+        json_file = BytesIO(json.dumps(json_data).encode())
+        taxonomy = tagging_api.create_taxonomy("Validations_Taxonomy")
+
+        with self.assertRaises(ValueError):
+            tagging_api.import_tags(taxonomy, json_file, tagging_api.TaxonomyDataFormat.JSON)
+
+    def test_import_tags_resync(self):
+        object_id = 'course_1'
+        object_tag = ObjectTag(
+            object_id=object_id,
+            object_type='course',
+            taxonomy=self.taxonomy,
+            tag=self.bacteria,
+        )
+        tagging_api.resync_object_tags([object_tag])
+        object_tag = ObjectTag.objects.get(object_id=object_id)
+        self.assertEqual(object_tag.tag.value, 'Bacteria')
+        self.assertEqual(object_tag._value, 'Bacteria')
```

Comment:
nit: pylint doesn't like this, but I agree it's necessary for the test. So could you please add this here and in the two other mentions below?

```suggestion
        self.assertEqual(object_tag._value, 'Bacteria')  # pylint: disable=protected-access
```